### ISSUE TITLE
[FEATURE REQ] mgmt Compute, convenience method for disk HyperVGeneration

### ISSUE THREAD

**Issue by XiaofeiCao** [Azure/azure-sdk-for-java#38163]
For managed disks, support HyperVGeneration in both create and update:
```java
Disk disk = disks
   .define(newOsDiskName)
   .withRegion(region)
   .withExistingResourceGroup(resourceGroupName)
   .withWindowsFromSnapshot(snapshot)
   .withSizeInGB(snapshot.sizeInGB())
   .withHyperVGeneration(HyperVGeneration.V1) // support HyperVGeneration in creation
   .createAsync().block()

disk.update()
  .withHyperVGeneration(HyperVGeneration.V2) // support update
  .apply()

```

similar to GalleryImage: https://github.com/Azure/azure-sdk-for-java/blob/74015974649f85214a9fe8c6f37dd92041d2e853/sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/SharedGalleryImageTests.java#L324-L335

**Comment by v-hongli1** [comment]
The fixed code have been merged into main branch.

**Comment by XiaofeiCao** [comment]
Will be available in 2.35.0, ETA 1/26.

**Comment by XiaofeiCao** [comment]
2.35.0 released.
